tests/test_unit/test_kinematics.py

```diff
@@ -1,112 +1,176 @@
             )
 
-        return _expected_dataarray
 
-    kinematic_test_params = [
-        ("valid_poses_dataset", does_not_raise()),
-        ("valid_poses_dataset_with_nan", does_not_raise()),
-        ("missing_dim_poses_dataset", pytest.raises(ValueError)),
-    ]
-
-    @pytest.mark.parametrize("ds, expected_exception", kinematic_test_params)
-    def test_displacement(
-        self, ds, expected_exception, expected_dataarray, request
-    ):
-        """Test displacement computation."""
-        ds = request.getfixturevalue(ds)
-        with expected_exception:
-            result = kinematics.compute_displacement(ds.position)
-            expected = expected_dataarray("displacement")
-            if ds.position.isnull().any():
-                expected.loc[
-                    {"individuals": "ind1", "time": [3, 4, 7, 8, 9]}
-                ] = np.nan
-            xr.testing.assert_allclose(result, expected)
-
-    @pytest.mark.parametrize("ds, expected_exception", kinematic_test_params)
-    def test_velocity(
-        self, ds, expected_exception, expected_dataarray, request
-    ):
-        """Test velocity computation."""
-        ds = request.getfixturevalue(ds)
-        with expected_exception:
-            result = kinematics.compute_velocity(ds.position)
-            expected = expected_dataarray("velocity")
-            if ds.position.isnull().any():
-                expected.loc[
-                    {"individuals": "ind1", "time": [2, 4, 6, 7, 8, 9]}
-                ] = np.nan
-            xr.testing.assert_allclose(result, expected)
-
-    @pytest.mark.parametrize("ds, expected_exception", kinematic_test_params)
-    def test_acceleration(
-        self, ds, expected_exception, expected_dataarray, request
-    ):
-        """Test acceleration computation."""
-        ds = request.getfixturevalue(ds)
-        with expected_exception:
-            result = kinematics.compute_acceleration(ds.position)
-            expected = expected_dataarray("acceleration")
-            if ds.position.isnull().any():
-                expected.loc[
-                    {"individuals": "ind1", "time": [1, 3, 5, 6, 7, 8, 9]}
-                ] = np.nan
-            xr.testing.assert_allclose(result, expected)
-
-    @pytest.mark.parametrize("order", [0, -1, 1.0, "1"])
-    def test_approximate_derivative_with_invalid_order(self, order):
-        """Test that an error is raised when the order is non-positive."""
-        data = np.arange(10)
-        expected_exception = (
-            ValueError if isinstance(order, int) else TypeError
+@pytest.mark.parametrize(
+    "valid_dataset_with_nan",
+    [
+        "valid_poses_dataset_with_nan",
+        "valid_bboxes_dataset_with_nan",
+    ],
+)
+@pytest.mark.parametrize(
+    "kinematic_variable, expected_nans_per_individual",
+    [
+        ("displacement", {0: 5, 1: 0}),
+        ("velocity", {0: 6, 1: 0}),
+        ("acceleration", {0: 7, 1: 0}),
+    ],
+)
```

Comment:
As above, I would use list instead of dict
```suggestion
@pytest.mark.parametrize(
    "kinematic_variable, expected_nans_per_individual",
    [
        ("displacement", [5, 0]),
        ("velocity", [6, 0]),
        ("acceleration", [7, 0]),
    ],
)
```